lib/modifiers.js

```diff
@@ -20,12 +20,15 @@ function groupRedact (o, path, censor, isCensorFct) {
   const target = get(o, path)
   if (target == null) return { keys: null, values: null, target: null, flat: true }
   const keys = Object.keys(target)
-  const length = keys.length
-  const values = new Array(length)
-  for (var i = 0; i < length; i++) {
-    const k = keys[i]
-    values[i] = target[k]
-    target[k] = isCensorFct ? censor(target[k]) : censor
+  const keysLength = keys.length
+  const pathLength = path.length
+  const pathWithKey = path.concat('')
+  const values = new Array(keysLength)
+  for (var i = 0; i < keysLength; i++) {
+    const key = keys[i]
+    pathWithKey[pathLength] = key
+    values[i] = target[key]
+    target[key] = isCensorFct ? censor(target[key], pathWithKey) : censor
```

Comment:
I would move the if check for isCensorFt outside the for loop, and avoid the pathWithKey allocation and manipulation if censor is not a function.

Reply:
Ok, I think I've addressed this and your comment below. Take a look!